src/main/kotlin/com/openlattice/data/storage/partitions/PartitionManager.kt

```diff
@@ -160,29 +120,15 @@ class PartitionManager @JvmOverloads constructor(
         }
     }
 
-    fun getPartitionInformation(): Map<Int, Long> {
-        return BasePostgresIterable(
-                PreparedStatementHolderSupplier(hds, ALL_PARTITIONS) { ps ->
-                    ps.setArray(1, PostgresArrays.createIntArray(ps.connection, partitionList))
-                }) { it.getInt(PARTITION.name) to it.getLong(COUNT) }.toMap()
-    }
-
-    fun allocateDefaultPartitions(partitionCount: Int): List<Int> {
-        return getEmptiestPartitions(partitionCount).map { it.first }
-    }
-
-    private fun getEmptiestPartitions(partitionCount: Int): BasePostgresIterable<Pair<Int, Long>> {
-        //A quick note is that the partitions materialized view shouldn't be turned into a distributed table
-        //with out addressing the interaction of the order by and limit clauses.
-        return BasePostgresIterable(
-                PreparedStatementHolderSupplier(hds, EMPTIEST_PARTITIONS) { ps ->
-                    ps.setArray(1, PostgresArrays.createIntArray(ps.connection, partitionList))
-                    ps.setInt(2, partitionCount)
-                }) { it.getInt(PARTITION.name) to it.getLong(COUNT) }
-    }
-
-    private fun isValidResize(partitionCount: Int) {
-        checkArgument(partitionCount > partitionList.size, "Only support resizing to a larger number of partitions.")
+    private fun getEmptiestPartitions(numPartitions: Int): List<Int> {
+        val partitionCounts = mutableMapOf<Int, Long>()
+        BasePostgresIterable(StatementHolderSupplier(hds, EMPTIEST_PARTITIONS)) {
+            ResultSetAdapters.count(it) to ResultSetAdapters.partitions(it)
+        }.forEach { (count, partitions) ->
+            val avg = count / partitions.size
```

Comment:
This seems to be computing emptiest partitions from a computation that assumes data is evenly spread across partitions for entity sets. We know the actual sizes from PostgresMaterializedViews for partition counts, why aren't we using that?

Reply:
It appears as though the previous get emptiest partitions call isn't actually working and creates a non-trivial performance impact whenever the counts get updated. 

I think assuming the data is evenly distributed should be fine since ekid -> partition mapping is random. Also I think this implementation should work about as well and end up reducing DB load.

The idea also is to add repartitioning to fix the ton of badly partitioned data 😢

Reply:
(repartitioning is l8r)